lambdas/commands.py

```diff
@@ -20,32 +20,47 @@ def generate_extract_command(
         source, run_date, run_type, step
     )
     extract_output_file = helpers.generate_step_output_filename(
-        "index", extract_output_prefix, step
+        source, "index", extract_output_prefix, step
     )
 
-    extract_command = [
-        f"--host={input_data['oai-pmh-host']}",
-        f"--output-file=s3://{timdex_bucket}/{extract_output_file}",
-    ]
+    extract_command = []
 
     if verbose:
         extract_command.append("--verbose")
 
-    extract_command.append("harvest")
-    extract_command.append(f"--metadata-format={input_data['oai-metadata-format']}")
-
-    if source in ["aspace", "dspace"]:
-        extract_command.append("--method=get")
+    if source in ["gismit", "gisogm"]:
+        extract_command.append("harvest")
+        if run_type == "daily":
+            extract_command.append("--harvest-type=incremental")
+            extract_command.append(
+                f"--from-date={helpers.generate_harvest_from_date(run_date)}"
+            )
+        elif run_type == "full":
+            extract_command.append("--harvest-type=full")
 
-    if set_spec := input_data.get("oai-set-spec"):
-        extract_command.append(f"--set-spec={set_spec}")
+        extract_command.append(
+            f"--output-file=s3://{timdex_bucket}/{extract_output_file}"
+        )
+        extract_command.append(source.removeprefix("gis"))
 
-    if run_type == "daily":
+    else:
+        extract_command.append(f"--host={input_data['oai-pmh-host']}")
         extract_command.append(
-            f"--from-date={helpers.generate_harvest_from_date(run_date)}",
+            f"--output-file=s3://{timdex_bucket}/{extract_output_file}"
         )
-    elif run_type == "full":
-        extract_command.append("--exclude-deleted")
+        extract_command.append("harvest")
+        if source in ["aspace", "dspace"]:
+            extract_command.append("--method=get")
+        extract_command.append(f"--metadata-format={input_data['oai-metadata-format']}")
+        if run_type == "daily":
+            extract_command.append(
+                f"--from-date={helpers.generate_harvest_from_date(run_date)}",
+            )
+        elif run_type == "full":
+            extract_command.append("--exclude-deleted")
+
+        if set_spec := input_data.get("oai-set-spec"):
+            extract_command.append(f"--set-spec={set_spec}")
```

Comment:
While I think these changes are very clean, easy to follow, and inline with the previous style of this function, what do you think about breaking the functionality of creating OAI and Geo harvester command generation into helper functions?  e.g. something like `generate_oai_harvester_extract_command()` and `generate_geo_harvester_extract_command()`?  They could live in the `helpers.py` file for the time being?

Ultimately, I'd imagine it would make them easier to test, though it would require some changes to the current tests.

Reply:
As discussed during our check-in, we decided to leave the code as-is for now and make additional improvements to the architecture (e.g., explore benefits of a source-level module) in the near future (potentially when we integrate `browsertrix-harvester` into TIMDEX).